src/indexer/utils/evm/index.ts

```diff
@@ -130,6 +145,35 @@ export function parseDeposit(
   }
 }
 
+function parseEvmDestination(bytes: Uint8Array): string {
+  const filtered = bytes.filter((_, idx) => idx + 1 > 65)
+  return hexlify(filtered)
+}
+
+async function ParseSubstrateDestination(bytes: Uint8Array, domain: Domain): Promise<string> {
+  const rpcUrlConfig = getSsmDomainConfig()
+  const wsProvider = new WsProvider(rpcUrlConfig.get(domain.id))
+  const api = await ApiPromise.create({
+    provider: wsProvider,
+  })
+
+  const recipientlen = Number("0x" + Buffer.from(bytes.slice(32, 64)).toString("hex"))
+
+  const recipient = "0x" + Buffer.from(bytes.slice(64, 64 + recipientlen)).toString("hex")
+
+  const decodedData = api.createType("MultiLocation", recipient)
+  const multiAddress = decodedData.toJSON() as unknown as MultiLocation
+
+  for (const [, junctions] of Object.entries(multiAddress.interior)) {
```

Comment:
Isn't there a function in polkadot.js lib to do this?

Reply:
I think not, at least i didn't find it